recipes/dotnet_agent.rb

```diff
@@ -29,32 +29,36 @@
 # Environment Validation
 
 # MSDTC Service
-service "MSDTC" do
-  action [ :enable, :start ]
+service 'MSDTC' do
+  action [:enable, :start]
 end
 
 # WMI Service
-service "Winmgmt" do
-  action [ :enable, :start ]
+service 'Winmgmt' do
+  action [:enable, :start]
 end
 
-if agent['version'] < '4.1'
-  # COM+ Service is not required in 4.1+
-  service "COMSysApp" do
-    action [ :enable, :start ]
-  end
+# COM+ Service is not required in 4.1+
+service 'COMSysApp' do
+  action [:enable, :start]
+  only_if { agent['version'] < '4.1.0.0' }
 end
 
 # Check whether IIS 7.0+ is installed
 # Enable IIS Health Monitoring for the Machine snapshots to return IIS App Pool data
 # There is no equivalent available in chef to get the IIS version - so completely using powershell scripts
 powershell_script 'check_IIS' do
-	code 'Install-WindowsFeature Web-Request-Monitor'
-	only_if '[Single]::Parse((get-itemproperty HKLM:\SOFTWARE\Microsoft\InetStp\  | select versionstring).VersionString.Substring(8)) -ge 7.0'
+  code 'Install-WindowsFeature Web-Request-Monitor'
+  only_if '[Single]::Parse((get-itemproperty HKLM:\SOFTWARE\Microsoft\InetStp\  | select versionstring).VersionString.Substring(8)) -ge 7.0'
+end
+
+# create directory if it doesn't exist
```

Comment:
The msi is stored on the Windows/Temp directory and it is done as the first step. @akemner - Why are we creating the directory here?